dynawo/sources/Modeler/Common/DYNConnector.cpp

```diff
@@ -147,16 +147,30 @@ ConnectorContainer::mergeFlowConnector() {
   // order flow connectors
   flowConnectors_.clear();
   flowConnectorByVarNum_.clear();
+  FlowAliasNameToFictitiousVarNum_.clear();
   list<shared_ptr<Connector> > flowConnectorsList;
   for (unsigned int i = 0; i < flowConnectorsDeclared_.size(); ++i) {
     shared_ptr<Connector> flowc(new Connector(*flowConnectorsDeclared_[i]));
     bool merged = false;
     for (vector<connectedSubModel>::iterator it = flowc->connectedSubModels().begin();
         it != flowc->connectedSubModels().end();
         ++it) {
-      const int numVar = it->subModel()->getVariableIndexGlobal(it->variable());
+      const boost::shared_ptr<Variable>& variable = it->variable();
+      int numVar;
+      if (variable->isAlias()) {
+        string id = it->subModel()->name()+"_"+variable->getName();
+        boost::unordered_map<std::string, int>::const_iterator aliasIt = FlowAliasNameToFictitiousVarNum_.find(id);
+        if (aliasIt != FlowAliasNameToFictitiousVarNum_.end()) {
+          numVar = aliasIt->second;
+        } else {
+          numVar = sizeY_ + FlowAliasNameToFictitiousVarNum_.size();
+          FlowAliasNameToFictitiousVarNum_[it->subModel()->name()+"_"+variable->getName()] = numVar;
+        }
+      } else {
+        numVar = it->subModel()->getVariableIndexGlobal(it->variable());
+      }
```

Comment:
We do need the first loop to check if this connector can be merged or not. Going through all connections to detect aliases and assign them an id would make a third loop, so I would prefer to keep assigning on the fly. However, I will add a method to determine the id of the alias so that the code is not duplicated.

Reply:
Solved